baseband/mark4/base.py

```diff
@@ -334,6 +343,16 @@ def _get_frame_rate(fh, header_template):
         tdelta = header1.ms[0] - header0.ms[0]
         return int(np.round(1000. / tdelta))
 
+    @lazyproperty
+    def _last_header(self):
+        """Last header of the file."""
+        last_header = super(Mark4StreamReader, self)._last_header
+        # decade correction.
+        header_unit_year = last_header['bcd_unit_year'][0]
+        last_header.decade = Mark4Header.infer_decade(self.header0.time,
```

Comment:
If we do not go with my suggestion, this should be changed to `last_header.infer_decade(...)`